### PR TITLE
Update for IPP for OpenCV 2017u2 integration;

### PR DESCRIPTION
**Pull request by lupustr3**
### This pullrequest changes

Updated integrations for:
cv::split
cv::merge
cv::insertChannel
cv::extractChannel
cv::Mat::convertTo - now with scaled conversions support
cv::LUT - disabled due to performance issues
Mat::copyTo
Mat::setTo
cv::flip
cv::copyMakeBorder - currently disabled
cv::polarToCart
cv::pow - ipp pow function was removed due to performance issues
cv::hal::magnitude32f/64f - disabled for <= SSE42, poor performance
cv::countNonZero
cv::minMaxIdx
cv::norm
cv::canny - new integration. Disabled for threaded;
cv::cornerHarris
cv::boxFilter
cv::bilateralFilter
cv::integral

### CODE REVIEW

**Comment by lupustr3** [comment]
I made test adjustment for bilateral filter in "test_bilateral_filter.cpp". Please check if it's ok.

File: modules/imgproc/test/test_bilateral_filter.cpp [hunk 1]
```diff
@@ -251,20 +251,23 @@ namespace cvtest
 
     int CV_BilateralFilterTest::validate_test_results(int test_case_index)
     {
-        static const double eps = 4;
-
+        double eps = (_src.depth() < CV_32F)?1:4e-5;
```
Comment (alalek) [inline modules/imgproc/test/test_bilateral_filter.cpp]: `4e-5` => `1e-3` to make ARM platform happy.

Comment (lupustr3) [inline modules/imgproc/test/test_bilateral_filter.cpp]: Thanks!

File: modules/imgproc/src/canny.cpp [hunk 2]
```diff
@@ -318,6 +316,8 @@ class parallelCanny : public ParallelLoopBody
         // In sobel transform we calculate ksize2 extra lines for the first and last rows of each slice
         // because IPPDerivSobel expects only isolated ROIs, in contrast with the opencv version which
```
Comment (lupustr3) [inline modules/imgproc/src/canny.cpp]: Note that current integration of IPP in Sobel supports in memory border and this part probably can be more efficient.

File: modules/imgproc/test/test_bilateral_filter.cpp (hunk 1 above)
Comment (lupustr3) [inline modules/imgproc/test/test_bilateral_filter.cpp]: It seems every next test is worse than previous for ARM accuracy )

**Comment by alalek** [comment]
Please fix `ippicv_2017u2_lnx_ia32` package name (date) [here](https://github.com/opencv/opencv/blob/41d55c5095c0db3f846b5ad05f001365c55e8687/3rdparty/ippicv/ippicv.cmake#L24): "ippicv_2017u2_lnx_ia32_20170418.tgz"

File: modules/imgproc/test/test_bilateral_filter.cpp (hunk 1 above)
Comment (alalek) [inline modules/imgproc/test/test_bilateral_filter.cpp]: So `5e-3` is the next limit.

**Comment by alalek** [comment]
I will take a look on Android builder problem.

**Comment by alalek** [comment]
Thank you! :+1:

**Pull request merged**